test/runTest.ts

```diff
@@ -0,0 +1,40 @@
+import * as path from 'path';
+import { runTests } from 'vscode-test';
+import { TestOptions } from 'vscode-test/out/runTest';
+
+async function main() {
+  try {
+    // The folder containing the Extension Manifest package.json
+    // Passed to `--extensionDevelopmentPath`
+    const extensionDevelopmentPath = path.resolve(__dirname, '../');
+
+    // The path to test runner
+    // Passed to --extensionTestsPath
+    const extensionTestsPath = path.resolve(__dirname, './suite/bootstrap');
+
+    // Set grep value for Mocha, to trigger only the type of tests passed in package.json
+    // format is `(type)`, where type = unit | integration | system
+    // if no type is passed, ALL tests are executed
+    const MOCHA_grep_value =
+      process.argv.length > 2 ? `(${process.argv[2]})` : '';
```

Comment:
This is used in the context of another launch action:

```
{
			"name": "Run Specific Test",
			"type": "extensionHost",
			"request": "launch",
			"runtimeExecutable": "${execPath}",
			"args": [
				"--disable-extensions",
				"--extensionDevelopmentPath=${workspaceFolder}",
				"--extensionTestsPath=${workspaceFolder}/test/suite/bootstrap"
      ],
      "env": {
        "MOCHA_grep": "${input:TestName}"
      },
			"outFiles": ["${workspaceFolder}/out/test/**/*.js"],
			"preLaunchTask": "npm: watch"
		}
  ],
```
and we need to add an input for this, that we can pass as an env variable.
```
  "inputs": [
    {
      "id": "TestName",
      "type": "promptString",
      "description": "Name of the test you want to run. Can be a substring.",
      "default": "in progress"
    }
  ]
```

This way, can can use any string in a `describe` of `it` statement, to run only that test.